openpgp/packet/private_key_test.go

```diff
@@ -56,6 +65,77 @@ func TestPrivateKeyRead(t *testing.T) {
 	}
 }
 
+func TestPrivateKeyEncrypt(t *testing.T) {
+	var encryptedMsg, decryptedMsg []byte
+	var encryptedC1, encryptedC2 *big.Int
+
+	for i, test := range privateKeyTests {
+		packet, err := Read(readerFromHex(test.privateKeyHex))
+		if err != nil {
+			t.Errorf("#%d: failed to parse: %s", i, err)
+			continue
+		}
+		privKey := packet.(*PrivateKey)
+		err = privKey.Decrypt([]byte("testing"))
+		if err != nil {
+			t.Errorf("#%d: failed to decrypt: %s", i, err)
+			continue
+		}
+
+		switch pubKey := privKey.PublicKey.PublicKey.(type) {
```

Comment:
I just tried to make it `privKey.Decrypt` after encrypting the message (so priv was still encrypted while it used pub to encrypt message) and it worked. Cool!

Maybe we could make it two tests, one with simple `priv.Decrypt; priv.Encrypt; priv.Decrypt` and some assertions that it worked, and another one with that roundtrip message encryption. What do you think?